bearer/bearer_test.go

```diff
@@ -165,235 +287,691 @@ func TestToken_InvalidAt(t *testing.T) {
+					145, 26, 229, 252, 61, 36, 160, 242, 243}},
+			{name: "body/subject/value/checksum mismatch", err: "invalid target user: checksum mismatch",
+				b: []byte{10, 29, 18, 27, 10, 25, 53, 147, 14, 186, 66, 195, 247, 51, 14, 249, 145, 102, 233, 115, 142, 143,
+					145, 26, 229, 252, 61, 36, 160, 242, 244}},
+			{name: "body/issuer/value/empty", err: "invalid issuer: invalid length 0, expected 25",
+				b: []byte{10, 2, 34, 0}},
+			{name: "body/issuer/value/undersize", err: "invalid issuer: invalid length 24, expected 25",
+				b: []byte{10, 28, 34, 26, 10, 24, 53, 147, 14, 186, 66, 195, 247, 51, 14, 249, 145, 102, 233, 115, 142, 143,
+					145, 26, 229, 252, 61, 36, 160, 242}},
+			{name: "body/issuer/value/oversize", err: "invalid issuer: invalid length 26, expected 25",
+				b: []byte{10, 30, 34, 28, 10, 26, 53, 147, 14, 186, 66, 195, 247, 51, 14, 249, 145, 102, 233, 115, 142, 143,
+					145, 26, 229, 252, 61, 36, 160, 242, 243, 1}},
+			{name: "body/issuer/value/wrong prefix", err: "invalid issuer: invalid prefix byte 0x42, expected 0x35",
+				b: []byte{10, 29, 34, 27, 10, 25, 66, 147, 14, 186, 66, 195, 247, 51, 14, 249, 145, 102, 233, 115, 142, 143,
+					145, 26, 229, 252, 61, 36, 160, 242, 243}},
+			{name: "body/issuer/value/checksum mismatch", err: "invalid issuer: checksum mismatch",
+				b: []byte{10, 29, 34, 27, 10, 25, 53, 147, 14, 186, 66, 195, 247, 51, 14, 249, 145, 102, 233, 115, 142, 143,
+					145, 26, 229, 252, 61, 36, 160, 242, 244}},
+			{name: "signature/invalid scheme", err: "invalid body signature: scheme 2147483648 overflows int32",
+				b: []byte{18, 11, 24, 128, 128, 128, 128, 248, 255, 255, 255, 255, 1}},
+		} {
+			t.Run(tc.name, func(t *testing.T) {
+				require.EqualError(t, new(bearer.Token).Unmarshal(tc.b), tc.err)
+			})
+		}
+	})
 
-	jd, err := token.MarshalJSON()
+	var val bearer.Token
+	// zero
+	require.NoError(t, val.Unmarshal(nil))
+
+	_, ok := val.GetEACLTable()
+	require.False(t, ok)
+	require.Zero(t, val.Issuer())
+	require.Zero(t, val.Exp())
+	require.Zero(t, val.Iat())
+	require.Zero(t, val.Nbf())
+	require.True(t, val.AssertContainer(cidtest.ID()))
+	require.True(t, val.AssertUser(usertest.ID()))
+	_, ok = val.Signature()
+	require.False(t, ok)
+
+	// filled
+	err := val.Unmarshal(validBinBearerToken)
 	require.NoError(t, err)
+	t.Skip("https://github.com/nspcc-dev/neofs-sdk-go/issues/606")
+	require.Equal(t, validBearerToken, val)
+}
 
-	require.NoError(t, val2.UnmarshalJSON(jd))
-
-	val2.WriteToV2(&msg)
-	require.Zero(t, msg.GetBody())
-
-	// set value
-	usr := usertest.ID()
-
-	var usrV2 refs.OwnerID
-	usr.WriteToV2(&usrV2)
-
-	token.SetIssuer(usr)
-
-	require.True(t, usr == token.Issuer())
-	require.True(t, usr == token.ResolveIssuer())
+func TestToken_MarshalJSON(t *testing.T) {
+	b, err := json.MarshalIndent(validBearerToken, "", " ")
+	require.NoError(t, err)
+	if string(b) != validJSONBearerToken {
+		// protojson is inconsistent https://github.com/golang/protobuf/issues/1121
```

Comment:
https://pkg.go.dev/github.com/stretchr/testify/require#JSONEq

Reply:
dont see how it can help me here

Reply:
done